apps/emqx_cluster_link/test/emqx_cluster_link_SUITE.erl

```diff
@@ -253,6 +255,61 @@ t_target_extrouting_gc(Config) ->
         Trace
     ).
 
+%% Checks that, if an exception occurs while handling a route op message, we disconnect
+%% the upstream agent client so it restarts.
+t_disconnect_on_errors('init', Config) ->
+    SourceNodes = emqx_cth_cluster:start(mk_source_cluster(?FUNCTION_NAME, Config)),
+    TargetNodes = emqx_cth_cluster:start(mk_target_cluster(?FUNCTION_NAME, Config)),
+    _Apps = start_cluster_link(SourceNodes ++ TargetNodes, Config),
+    ok = snabbkaffe:start_trace(),
+    [
+        {source_nodes, SourceNodes},
+        {target_nodes, TargetNodes}
+        | Config
+    ];
+t_disconnect_on_errors('end', Config) ->
+    ok = snabbkaffe:stop(),
+    ok = emqx_cth_cluster:stop(?config(source_nodes, Config)),
+    ok = emqx_cth_cluster:stop(?config(target_nodes, Config)).
+t_disconnect_on_errors(Config) ->
+    ct:timetrap({seconds, 20}),
+    [SN1 | _] = nodes_source(Config),
+    TargetNodes = nodes_target(Config),
+    SC1 = start_client("t_disconnect_on_errors", SN1),
+    ok = ?ON(SN1, meck:new(emqx_cluster_link, [passthrough, no_link, no_history])),
+    ?assertMatch(
+        {_, {ok, _}},
+        ?wait_async_action(
+            begin
+                lists:foreach(
+                    fun(N) ->
+                        ok = ?ON(
+                            N,
+                            meck:expect(
+                                emqx_cluster_link,
+                                do_handle_route_op_msg,
+                                fun(_Msg) ->
+                                    meck:exception(error, {badmatch, []})
+                                end
+                            )
+                        )
+                    end,
+                    TargetNodes
+                ),
```

Comment:
Nit. Probably not too important to have more than 1 node in the target cluster for this testcase. Worth simplifying by using single-node cluster?